__tests__/unit/backend-proxy.test.js

```diff
@@ -282,5 +282,97 @@ describe('Backend Proxy', () => {
 				message: expect.stringContaining('Unexpected token')
 			}));
 		});
+
+		test('reads the backend response and makes the location a relative URL if the domain is the backend', () => {
+			// Given
+			const middleware = backendProxy(baseOptions);
+			let relativeLocation = `/some-wonderful/location?here=there#my-id`;
+
+			const backendResponse = new EventEmitter();
+			backendResponse.headers = {
+				location: baseOptions.backend + relativeLocation
+			};
+
+			backendResponse.statusCode = 302;
+			backendResponse.pipe = jest.fn();
+
+			const response = {
+				header: jest.fn()
+			};
+
+			middleware(mockRequest, response, next);
+
+			// When
+			proxyRequest.emit('response', backendResponse);
+
+			// Then
+			expect(backendResponse.pipe).toHaveBeenCalledTimes(1);
+			expect(backendResponse.pipe).toHaveBeenCalledWith(response);
+			expect(response.header).toHaveBeenCalledWith({
+				...backendResponse.headers,
+				location: relativeLocation
+			});
+			expect(response.statusCode).toBe(backendResponse.statusCode);
+			expect(next).not.toHaveBeenCalled();
+		});
+
+		test(`reads the backend response and doesn't add a location is one isn't present`, () => {
+			// Given
+			const middleware = backendProxy(baseOptions);
+
+			const backendResponse = new EventEmitter();
+			backendResponse.headers = {
+				'content-type': 'text/plain'
+			};
+
+			backendResponse.statusCode = 302;
```

Comment:
Worth using a status code of `300` and then one of `399`?

Reply:
done